### PR TITLE
example/copy supports vp9-in-mp4 and example/dump supports fmp4

### PR DESCRIPTION
**Pull request by nintha**

### CODE REVIEW

**Comment by alfg** [comment]
Thank you @nintha! I will check this out tonight.

File: src/types.rs [hunk 1]
```diff
@@ -287,18 +293,23 @@ impl Into<&str> for &MediaType {
...
+    // 66 with constraint set 1
+    AvcBaseline,
+    // 66,
+    AvcMain,
+    // 77,
+    AvcExtended,
+    // 88,
     AvcHigh,                // 100
-                            // TODO Progressive High Profile, Constrained High Profile, ...
+    // TODO Progressive High Profile, Constrained High Profile, ...
```
Comment (alfg) [inline src/types.rs:312]: Is this an auto reformatting thing?

File: src/types.rs [hunk 2]
```diff
@@ -287,18 +293,23 @@ impl Into<&str> for &MediaType {
             MediaType::H265 => MEDIA_TYPE_H265,
             MediaType::AAC => MEDIA_TYPE_AAC,
             MediaType::TTXT => MEDIA_TYPE_TTXT,
+            MediaType::VP9 => MEDIA_TYPE_VP9,
```
Comment (alfg) [inline src/types.rs:296]: Can this match the ordering from above? Right after H265 type.

File: src/types.rs [hunk 3]
```diff
@@ -497,12 +508,19 @@ impl Default for AacConfig {
...
+    pub height: u16,
+}
+
 #[derive(Debug, PartialEq, Clone)]
 pub enum MediaConfig {
     AvcConfig(AvcConfig),
     HevcConfig(HevcConfig),
     AacConfig(AacConfig),
     TtxtConfig(TtxtConfig),
+    Vp9Config(Vp9Config),
```
Comment (alfg) [inline src/types.rs:523]: Add after the HevcConfig.

**Review by alfg — CHANGES_REQUESTED** [review]
Hey @nintha,

Thanks for adding VP9 support! I have a few comments/suggestions:

I encoded a sample mp4 using Tears of Steel:
```
ffmpeg -i tos.mp4 -c:v libvpx-vp9 -c:a copy tos-vp9.mp4
```

However, when running the `mp4info` and `simple` example, I am getting the following error:
```
Found 1 Tracks
unsupported sample entry box
```

It seems VP9 support also needs to be added to the following areas:
https://github.com/alfg/mp4-rust/blob/master/src/track.rs#L112-L138

I also added a few comments around formatting.

Thanks!

File: src/types.rs (hunk 1 above)
Comment (nintha) [reply src/types.rs:312]: Yes, I use IDEA auto reformatting. I will undo it to reduce unnecessary changes.

File: src/types.rs (hunk 2 above)
Comment (nintha) [reply src/types.rs:296]: Sure.

File: src/types.rs (hunk 3 above)
Comment (nintha) [reply src/types.rs:523]: Sure. I will fix it.

**Comment by nintha** [comment]
Hi @alfg ,

Thanks for your  comments/suggestions. I already undo the unnecessary changes and add vp9 support in `track.rs`.

I tested `mp4info` and `simple` with a mp4 file that uploaded as a attchment.

https://user-images.githubusercontent.com/22030293/105712401-b1cfa880-5f54-11eb-9f92-2a31495bb709.mp4

**Comment by alfg** [comment]
Great. Thanks for fixing. I'll check this out tonight.

**Review by alfg — CHANGES_REQUESTED** [review]
Hey @nintha, 

Thanks for adding the changes. Everything almost looks good, except I'm noticing one issue:

I'm seeing an issue when using `mp4copy` on the [minimal.mp4](https://github.com/alfg/mp4-rust/blob/master/tests/samples/minimal.mp4) example encoded with vp9. 

For example:
```
ffmpeg -i minimal.mp4 -c:v libvpx-vp9 -c:a copy minimal-vp9.mp4
cargo run --example mp4copy minimal-vp9.mp4 minimal-vp9-copy.mp4
ffplay minimal-vp9-copy.mp4
```

I see the following error. It also does not play in VLC either.
```
[mov,mp4,m4a,3gp,3g2,mj2 @ 00000228c2b781c0] overread end of atom 'stsd' by 86 bytes
[mov,mp4,m4a,3gp,3g2,mj2 @ 00000228c2b781c0] Could not find codec parameters for stream 0 (Video: none (stts / 0x73747473), none, 7 kb/s): unknown codec
Consider increasing the value for the 'analyzeduration' and 'probesize' options
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'minimal-vp9-copy.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2mp41
  Duration: 00:00:00.06, start: 0.000000, bitrate: 204 kb/s
    Stream #0:0(und): Video: none (stts / 0x73747473), none, 7 kb/s, 25 fps, 25 tbr, 12800 tbn, 12800 tbc (default)
    Metadata:
      encoder         : c
    Side data:
      displaymatrix: rotation of nan degrees
    Stream #0:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 48000 Hz, mono, fltp, 67 kb/s (default)
No decoder could be found for codec none
   0.04 M-A:  0.000 fd=   0 aq=    0KB vq=    0KB sq=    0B f=0/0
```

However, using your `ad_vp9.mp4` example seems to work OK. Any ideas? I've attached my copy of the `minimal-vp9.mp4` example for reference.

Other than that, everything else works good!

https://user-images.githubusercontent.com/702541/105791440-f3ebf080-5f3a-11eb-82c0-6bf1ac91b59a.mp4

**Comment by nintha** [comment]
Hi, @alfg 

I viewed the `minimal-vp9-copy.mp4` file in a hex viewer, and found nothing in the `stsd` box . So `ffplay` said "not find codec".

Now, I add some codes about appending `vp09` box into `stsd` box in the struct `Mp4TrackWriter`. 
https://github.com/nintha/mp4-rust/blob/a0265654f1258a070f08fb6bc8a62afd9fadd75f/src/track.rs#L575-L579

The `ffplay` works well with `minimal-vp9-copy.mp4` after fixing.

Thx.

**Review by alfg — APPROVED** [review]

**Comment by alfg** [comment]
Looks good. Thanks for all the work you put into this! I'll be publishing a new version shortly.

**Comment by alfg** [comment]
https://crates.io/crates/mp4/0.8.0

**Pull request merged**